In [7]:
# Importación de bibliotecas.
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
print('listo')

listo


In [8]:
# Configuración de estilo para los gráficos.
sns.set(style="whitegrid", palette="muted")
plt.rcParams["figure.figsize"] = (8,5)

## 1. Cargar el dataset.

Primero cargaremos los csv correspondientes de forma separada.

In [9]:
RAW = "../data/raw/"

# Cargas con parse_dates donde corresponde
orders = pd.read_csv(
    RAW + "olist_orders_dataset.csv",
    parse_dates=[
        "order_purchase_timestamp",
        "order_approved_at",
        "order_delivered_carrier_date",
        "order_delivered_customer_date",
        "order_estimated_delivery_date",
    ]
)

order_items = pd.read_csv(
    RAW + "olist_order_items_dataset.csv",
    parse_dates=["shipping_limit_date"]  # fecha límite de envío
)

order_payments = pd.read_csv(RAW + "olist_order_payments_dataset.csv")

order_reviews = pd.read_csv(
    RAW + "olist_order_reviews_dataset.csv",
    parse_dates=["review_creation_date", "review_answer_timestamp"]
)

customers = pd.read_csv(RAW + "olist_customers_dataset.csv")
products  = pd.read_csv(RAW + "olist_products_dataset.csv")
sellers   = pd.read_csv(RAW + "olist_sellers_dataset.csv")
cat_tr    = pd.read_csv(RAW + "product_category_name_translation.csv")



## 2. Analisis exploratorio inicial de los datos EDA 

Para este analisis se realizaran todos los siguientes puntos esperando lograr claridad sobre el dataset completo, se recalca que cada punto se realizara de forma separa para cada df 

- Cargar las 5 primeras filas de cada DataFrame.
- Revisar la estructura del DataFrame.
- Revisar los tipos de datos.
- Aplicar estadística descriptiva a cada una de la columnas del DataFrame:
  - Media (promedio).
  - Mediana.
  - Moda.
  - Desviación estándar.
  - Máximo.
  - Mínimo.
  - Rango.
  - Cuartiles.
  - Asimetría.
  - Curtosis. 
- Verificar valores nulos, registros faltantes y duplicados.
- Verificar valores extremos (outliers).

### Cargar las 5 primeras filas de cada DataFrame.

In [48]:
# Primeras 5 filas del DataFrame orders
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: orders")
print("=" * 70)
orders.head()

PRIMERAS 5 FILAS DEL DATAFRAME: orders


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26


In [49]:
# Primeras 5 filas del DataFrame order_items
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: order_items")
print("=" * 70)
order_items.head()

PRIMERAS 5 FILAS DEL DATAFRAME: order_items


,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14


In [50]:
# Primeras 5 filas del DataFrame order_payments
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: order_payments")
print("=" * 70)
order_payments.head()

PRIMERAS 5 FILAS DEL DATAFRAME: order_payments


,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


In [7]:
# Primeras 5 filas del DataFrame order_reviews
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: order_reviews")
print("=" * 70)
order_reviews.head()

PRIMERAS 5 FILAS DEL DATAFRAME: order_reviews


,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
0,7bc2406110b926393aa56f80a40eba40,73fc7af87114b39712e6da79b0a377eb,4,NaN,NaN,2018-01-18,2018-01-18 21:46:59
1,80e641a11e56f04c1ad469d5645fdfde,a548910a1c6147796b98fdf73dbeba33,5,NaN,NaN,2018-03-10,2018-03-11 03:05:13
2,228ce5500dc1d8e020d8d1322874b6f0,f9e4b658b201a9f2ecdecbb34bed034b,5,NaN,NaN,2018-02-17,2018-02-18 14:36:24
3,e64fb393e7b32834bb789ff8bb30750e,658677c97b385a9be170737859d3511b,5,NaN,Recebi bem antes do prazo estipulado.,2017-04-21,2017-04-21 22:02:06
4,f7c4243c7fe1938f181bec41a392bdeb,8e6bfb81e283fa7e4f11123a3fb894f1,5,NaN,Parabéns lojas lannister adorei comprar pela I...,2018-03-01,2018-03-02 10:26:53


In [8]:
# Primeras 5 filas del DataFrame customers
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: customers")
print("=" * 70)
customers.head()

PRIMERAS 5 FILAS DEL DATAFRAME: customers


,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


In [9]:
# Primeras 5 filas del DataFrame products
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: products")
print("=" * 70)
products.head()

PRIMERAS 5 FILAS DEL DATAFRAME: products


,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0


In [10]:
# Primeras 5 filas del DataFrame sellers
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: sellers")
print("=" * 70)
sellers.head()

PRIMERAS 5 FILAS DEL DATAFRAME: sellers


,seller_id,seller_zip_code_prefix,seller_city,seller_state
0,3442f8959a84dea7ee197c632cb2df15,13023,campinas,SP
1,d1b65fc7debc3361ea86b5f14c68d2e2,13844,mogi guacu,SP
2,ce3ad9de960102d0677a81f5d0bb7b2d,20031,rio de janeiro,RJ
3,c0f3eea2e14555b6faeea3dd58c1b1c3,4195,sao paulo,SP
4,51a04a8a6bdcb23deccc82b0b80742cf,12914,braganca paulista,SP


In [11]:
# Primeras 5 filas del DataFrame cat_tr
print("=" * 70)
print("PRIMERAS 5 FILAS DEL DATAFRAME: cat_tr")
print("=" * 70)
cat_tr.head()

PRIMERAS 5 FILAS DEL DATAFRAME: cat_tr


,product_category_name,product_category_name_english
0,beleza_saude,health_beauty
1,informatica_acessorios,computers_accessories
2,automotivo,auto
3,cama_mesa_banho,bed_bath_table
4,moveis_decoracao,furniture_decor


### Revisar la estructura de cada DataFrame.

In [10]:
# Estructura del DataFrame orders
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: orders")
print("=" * 50)
orders.info()

ESTRUCTURA DEL DATAFRAME: orders
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   order_id                       99441 non-null  object        
 1   customer_id                    99441 non-null  object        
 2   order_status                   99441 non-null  object        
 3   order_purchase_timestamp       99441 non-null  datetime64[ns]
 4   order_approved_at              99281 non-null  datetime64[ns]
 5   order_delivered_carrier_date   97658 non-null  datetime64[ns]
 6   order_delivered_customer_date  96476 non-null  datetime64[ns]
 7   order_estimated_delivery_date  99441 non-null  datetime64[ns]
dtypes: datetime64[ns](5), object(3)
memory usage: 6.1+ MB


El DataFrame orders reúne 99,441 pedidos con 8 columnas, por lo que es la fuente principal para analizar el ciclo de vida de la orden. La cobertura de fechas es alta pero no perfecta, ya que faltan 160 valores en order_approved_at, 783 en order_delivered_carrier_date y 2,965 en order_delivered_customer_date, mientras que order_estimated_delivery_date está completo. En la práctica, esto implica calcular métricas como delivery_days y % on_time únicamente cuando existan tanto la fecha de entrega real como la estimada. 

In [ ]:
# Estructura del DataFrame order_items
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: order_items")
print("=" * 50)
order_items.info()

ESTRUCTURA DEL DATAFRAME: order_items
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype         
---  ------               --------------   -----         
 0   order_id             112650 non-null  object        
 1   order_item_id        112650 non-null  int64         
 2   product_id           112650 non-null  object        
 3   seller_id            112650 non-null  object        
 4   shipping_limit_date  112650 non-null  datetime64[ns]
 5   price                112650 non-null  float64       
 6   freight_value        112650 non-null  float64       
dtypes: datetime64[ns](1), float64(2), int64(1), object(3)
memory usage: 6.0+ MB


El DataFrame order_items contiene 112,650 registros y 7 columnas sin valores nulos, con tipos correctos como fecha en shipping_limit_date, numéricos en price y freight_value, entero para order_item_id. Representa el detalle por ítem y varias filas por order_id, por lo que la clave práctica a validar es order_id, order_item_id. 

In [ ]:
# Estructura del DataFrame order_payments
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: order_payments")
print("=" * 50)
order_payments.info()

ESTRUCTURA DEL DATAFRAME: order_payments
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


El DataFrame order_payments reúne 103,886 registros y 5 columnas sin nulos, con múltiples filas por order_id debido a pagos parciales/secuenciales (payment_sequential) y distintos payment_type. Los campos numéricos payment_installments y payment_value permiten derivar a nivel orden la suma pagada, # de pagos y cuotas máximas. 

In [ ]:
# Estructura del DataFrame order_reviews
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: order_reviews")
print("=" * 50)
order_reviews.info()

ESTRUCTURA DEL DATAFRAME: order_reviews
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   review_id                99224 non-null  object        
 1   order_id                 99224 non-null  object        
 2   review_score             99224 non-null  int64         
 3   review_comment_title     11568 non-null  object        
 4   review_comment_message   40977 non-null  object        
 5   review_creation_date     99224 non-null  datetime64[ns]
 6   review_answer_timestamp  99224 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(1), object(4)
memory usage: 5.3+ MB


El DataFrame order_reviews aporta 99,224 reseñas con 7 columnas, las fechas review_creation_date y review_answer_timestamp, y los campos clave review_id, order_id, review_score tienen 100% de cobertura.

In [ ]:
# Estructura del DataFrame customers
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: customers")
print("=" * 50)
customers.info()

ESTRUCTURA DEL DATAFRAME: customers
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


El DataFrame customers tiene 99,441 filas y 5 columnas sin nulos, con customer_id, customer_unique_id, y la ubicación con city, state, zip_code_prefix. Es habitual que un mismo customer_unique_id aparezca en varios customer_id, ya que un cliente real puede tener más de un identificador operacional, y esto es útil para analizar recompra y frecuencia si luego agregamos por customer_unique_id. 

In [ ]:
# Estructura del DataFrame products
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: products")
print("=" * 50)
products.info()

ESTRUCTURA DEL DATAFRAME: products
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB


El DataFrame products reúne 32,951 productos con 9 columnas y estructura enfocada en dos bloques, atributos de contenido con product_category_name, product_name_lenght, product_description_lenght, y product_photos_qty.  Dimensiones físicas del producto con product_weight_g, product_length_cm,  product_height_cm y product_width_cm.  La columna product_id funciona como clave primaria y es la columna para enlazar con order_items.


In [ ]:
# Estructura del DataFrame sellers
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: sellers")
print("=" * 50)
sellers.info()

ESTRUCTURA DEL DATAFRAME: sellers
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


El DataFrame sellers reúne 3,095 vendedores con 4 columnas sin nulos, lo que indica buena calidad básica para cortes geográficos (seller_city, seller_state). Esta tabla será la clave para segmentar desempeño de vendedores (volumen, rapidez y reseñas) al unirse con order_items y, vía orders, con los hitos logísticos.

In [ ]:
# Estructura del DataFrame cat_tr
print("=" * 50)
print("ESTRUCTURA DEL DATAFRAME: cat_tr")
print("=" * 50)
cat_tr.info()

ESTRUCTURA DEL DATAFRAME: cat_tr
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB


El DataFrame cat_tr contiene 71 filas y 2 columnas sin nulos. Esta tabla es pequeña, estable y suficiente para rotular gráficos/tablas durante el análisis.

### Revisar los tipos de datos.

In [20]:
# Tipos de datos del DataFrame orders
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: orders")
print("=" * 50)
print(orders.dtypes)

TIPOS DE DATOS DEL DATAFRAME: orders
order_id                                 object
customer_id                              object
order_status                             object
order_purchase_timestamp         datetime64[ns]
order_approved_at                datetime64[ns]
order_delivered_carrier_date     datetime64[ns]
order_delivered_customer_date    datetime64[ns]
order_estimated_delivery_date    datetime64[ns]
dtype: object


In [21]:
# Tipos de datos del DataFrame order_items
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: order_items")
print("=" * 50)
print(order_items.dtypes)

TIPOS DE DATOS DEL DATAFRAME: order_items
order_id                       object
order_item_id                   int64
product_id                     object
seller_id                      object
shipping_limit_date    datetime64[ns]
price                         float64
freight_value                 float64
dtype: object


In [22]:
# Tipos de datos del DataFrame order_payments
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: order_payments")
print("=" * 50)
print(order_payments.dtypes)

TIPOS DE DATOS DEL DATAFRAME: order_payments
order_id                 object
payment_sequential        int64
payment_type             object
payment_installments      int64
payment_value           float64
dtype: object


In [23]:
# Tipos de datos del DataFrame order_reviews
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: order_reviews")
print("=" * 50)
print(order_reviews.dtypes)

TIPOS DE DATOS DEL DATAFRAME: order_reviews
review_id                          object
order_id                           object
review_score                        int64
review_comment_title               object
review_comment_message             object
review_creation_date       datetime64[ns]
review_answer_timestamp    datetime64[ns]
dtype: object


In [24]:
# Tipos de datos del DataFrame customers
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: customers")
print("=" * 50)
print(customers.dtypes)

TIPOS DE DATOS DEL DATAFRAME: customers
customer_id                 object
customer_unique_id          object
customer_zip_code_prefix     int64
customer_city               object
customer_state              object
dtype: object


In [25]:
# Tipos de datos del DataFrame products
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: products")
print("=" * 50)
print(products.dtypes)

TIPOS DE DATOS DEL DATAFRAME: products
product_id                     object
product_category_name          object
product_name_lenght           float64
product_description_lenght    float64
product_photos_qty            float64
product_weight_g              float64
product_length_cm             float64
product_height_cm             float64
product_width_cm              float64
dtype: object


In [26]:
# Tipos de datos del DataFrame sellers
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: sellers")
print("=" * 50)
print(sellers.dtypes)

TIPOS DE DATOS DEL DATAFRAME: sellers
seller_id                 object
seller_zip_code_prefix     int64
seller_city               object
seller_state              object
dtype: object


In [27]:
# Tipos de datos del DataFrame cat_tr
print("=" * 50)
print("TIPOS DE DATOS DEL DATAFRAME: cat_tr")
print("=" * 50)
print(cat_tr.dtypes)

TIPOS DE DATOS DEL DATAFRAME: cat_tr
product_category_name            object
product_category_name_english    object
dtype: object


### Aplicar estadística descriptiva a cada una de la columnas de cada DataFrame

In [28]:
# Estadística descriptiva del DataFrame order_items
print("=" * 70)
print("ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: order_items")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_order_items = order_items.select_dtypes(include=[np.number])

# Estadísticas descriptivas básicas
print("\n--- Estadísticas Básicas ---")
print(numeric_columns_order_items.describe())

# Estadísticas adicionales
print("\n--- Estadísticas Adicionales ---")
print(f"Media:\n{numeric_columns_order_items.mean()}\n")
print(f"Mediana:\n{numeric_columns_order_items.median()}\n")
print(f"Moda:\n{numeric_columns_order_items.mode().iloc[0]}\n")
print(f"Desviación estándar:\n{numeric_columns_order_items.std()}\n")
print(f"Mínimo:\n{numeric_columns_order_items.min()}\n")
print(f"Máximo:\n{numeric_columns_order_items.max()}\n")
print(f"Rango:\n{numeric_columns_order_items.max() - numeric_columns_order_items.min()}\n")
print(f"Asimetría (Skewness):\n{numeric_columns_order_items.skew()}\n")
print(f"Curtosis:\n{numeric_columns_order_items.kurtosis()}\n")

ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: order_items

--- Estadísticas Básicas ---
       order_item_id          price  freight_value
count  112650.000000  112650.000000  112650.000000
mean        1.197834     120.653739      19.990320
std         0.705124     183.633928      15.806405
min         1.000000       0.850000       0.000000
25%         1.000000      39.900000      13.080000
50%         1.000000      74.990000      16.260000
75%         1.000000     134.900000      21.150000
max        21.000000    6735.000000     409.680000

--- Estadísticas Adicionales ---
Media:
order_item_id      1.197834
price            120.653739
freight_value     19.990320
dtype: float64

Mediana:
order_item_id     1.00
price            74.99
freight_value    16.26
dtype: float64

Moda:
order_item_id     1.0
price            59.9
freight_value    15.1
Name: 0, dtype: float64

Desviación estándar:
order_item_id      0.705124
price            183.633928
freight_value     15.806405
dtype: float64

Mínimo:

En order_items se nota una cola larga, hay pocos ítems con precios y fletes muy altos con máx alrededor de $6,735 y $409.68 que elevan el promedio, mientras que el comportamiento típico es mucho menor con mediana con un precio alrededor $74.99, y freight_value alrededor de $16.26. El order_item_id casi siempre es 1, lo que sugiere pedidos con pocos ítems y algunos casos puntuales con muchos. La asimetría y curtosis son muy altas en price y freight_value, confirmando que la distribución está dominada por valores extremos y que la mediana describe mejor el centro que la media. También aparece freight_value en 0.00, indicio de registros con costo de envío nulo.

In [29]:
# Estadística descriptiva del DataFrame order_payments
print("=" * 70)
print("ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: order_payments")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_order_payments = order_payments.select_dtypes(include=[np.number])

# Estadísticas descriptivas básicas
print("\n--- Estadísticas Básicas ---")
print(numeric_columns_order_payments.describe())

# Estadísticas adicionales
print("\n--- Estadísticas Adicionales ---")
print(f"Media:\n{numeric_columns_order_payments.mean()}\n")
print(f"Mediana:\n{numeric_columns_order_payments.median()}\n")
print(f"Moda:\n{numeric_columns_order_payments.mode().iloc[0]}\n")
print(f"Desviación estándar:\n{numeric_columns_order_payments.std()}\n")
print(f"Mínimo:\n{numeric_columns_order_payments.min()}\n")
print(f"Máximo:\n{numeric_columns_order_payments.max()}\n")
print(f"Rango:\n{numeric_columns_order_payments.max() - numeric_columns_order_payments.min()}\n")
print(f"Asimetría (Skewness):\n{numeric_columns_order_payments.skew()}\n")
print(f"Curtosis:\n{numeric_columns_order_payments.kurtosis()}\n")

ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: order_payments

--- Estadísticas Básicas ---
       payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000

--- Estadísticas Adicionales ---
Media:
payment_sequential        1.092679
payment_installments      2.853349
payment_value           154.100380
dtype: float64

Mediana:
payment_sequential        1.0
payment_installments      1.0
payment_value           100.0
dtype: float64

Moda:
payment_sequential       1.0
payment_installments     1.0
payment_

En order_payments se observa que la mayoría de pedidos tiene un solo pago con payment_sequential, mediana=1 y moda=1, aunque existen casos puntuales con muchos pagos con máx = 29. Las cuotas suelen ser 1 ya que la mediana=1, pero hay variedad hasta 24, lo que indica comportamientos de financiamiento distintos. El monto de pago muestra cola larga, con mediana de $100, con valores muy altos con un máx de alrededor de $13,664 que elevan el promedio. También aparecen montos en 0 y una moda en $50, señal de pagos pequeños frecuentes. En conjunto, las distribuciones están muy concentradas en valores bajos con pocos casos extremos que dominan la cola.

In [30]:
# Estadística descriptiva del DataFrame order_reviews
print("=" * 70)
print("ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: order_reviews")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_order_reviews = order_reviews.select_dtypes(include=[np.number])

# Estadísticas descriptivas básicas
print("\n--- Estadísticas Básicas ---")
print(numeric_columns_order_reviews.describe())

# Estadísticas adicionales
print("\n--- Estadísticas Adicionales ---")
print(f"Media:\n{numeric_columns_order_reviews.mean()}\n")
print(f"Mediana:\n{numeric_columns_order_reviews.median()}\n")
print(f"Moda:\n{numeric_columns_order_reviews.mode().iloc[0]}\n")
print(f"Desviación estándar:\n{numeric_columns_order_reviews.std()}\n")
print(f"Mínimo:\n{numeric_columns_order_reviews.min()}\n")
print(f"Máximo:\n{numeric_columns_order_reviews.max()}\n")
print(f"Rango:\n{numeric_columns_order_reviews.max() - numeric_columns_order_reviews.min()}\n")
print(f"Asimetría (Skewness):\n{numeric_columns_order_reviews.skew()}\n")
print(f"Curtosis:\n{numeric_columns_order_reviews.kurtosis()}\n")

ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: order_reviews

--- Estadísticas Básicas ---
       review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

--- Estadísticas Adicionales ---
Media:
review_score    4.086421
dtype: float64

Mediana:
review_score    5.0
dtype: float64

Moda:
review_score    5
Name: 0, dtype: int64

Desviación estándar:
review_score    1.347579
dtype: float64

Mínimo:
review_score    1
dtype: int64

Máximo:
review_score    5
dtype: int64

Rango:
review_score    4
dtype: int64

Asimetría (Skewness):
review_score   -1.364386
dtype: float64

Curtosis:
review_score    0.507653
dtype: float64



Las reseñas en order_reviews están concentradas en calificaciones altasm con una mediana de 5 y el 75% de los casos están en 4–5 (Q1=4, Q3=5). Aun así, existe una cola hacia las bajas con un mínimo=1, reflejada en la asimetría negativa y en una desviación estándar  de 1.35, lo que indica una minoría de experiencias negativas que conviene tener presente. En promedio, la nota es 4.09, con moda en 5, señal de que la mayoría de clientes evalúa positivamente, aunque hay dispersión suficiente para distinguir diferencias de satisfacción entre grupos.

In [31]:
# Estadística descriptiva del DataFrame customers
print("=" * 70)
print("ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: customers")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_customers = customers.select_dtypes(include=[np.number])

if numeric_columns_customers.empty:
    print("\nNo hay columnas numéricas en este DataFrame.")
else:
    # Estadísticas descriptivas básicas
    print("\n--- Estadísticas Básicas ---")
    print(numeric_columns_customers.describe())
    
    # Estadísticas adicionales
    print("\n--- Estadísticas Adicionales ---")
    print(f"Media:\n{numeric_columns_customers.mean()}\n")
    print(f"Mediana:\n{numeric_columns_customers.median()}\n")
    print(f"Moda:\n{numeric_columns_customers.mode().iloc[0]}\n")
    print(f"Desviación estándar:\n{numeric_columns_customers.std()}\n")
    print(f"Mínimo:\n{numeric_columns_customers.min()}\n")
    print(f"Máximo:\n{numeric_columns_customers.max()}\n")
    print(f"Rango:\n{numeric_columns_customers.max() - numeric_columns_customers.min()}\n")
    print(f"Asimetría (Skewness):\n{numeric_columns_customers.skew()}\n")
    print(f"Curtosis:\n{numeric_columns_customers.kurtosis()}\n")

ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: customers

--- Estadísticas Básicas ---
       customer_zip_code_prefix
count              99441.000000
mean               35137.474583
std                29797.938996
min                 1003.000000
25%                11347.000000
50%                24416.000000
75%                58900.000000
max                99990.000000

--- Estadísticas Adicionales ---
Media:
customer_zip_code_prefix    35137.474583
dtype: float64

Mediana:
customer_zip_code_prefix    24416.0
dtype: float64

Moda:
customer_zip_code_prefix    22790
Name: 0, dtype: int64

Desviación estándar:
customer_zip_code_prefix    29797.938996
dtype: float64

Mínimo:
customer_zip_code_prefix    1003
dtype: int64

Máximo:
customer_zip_code_prefix    99990
dtype: int64

Rango:
customer_zip_code_prefix    98987
dtype: int64

Asimetría (Skewness):
customer_zip_code_prefix    0.779025
dtype: float64

Curtosis:
customer_zip_code_prefix   -0.788204
dtype: float64



En customers, el zip_code_prefix abarca un rango muy amplio con alrededor 1,003 a 99,990, lo que sugiere cobertura geográfica extendida. La mediana de 24,416 está por debajo de la media 35,137, con asimetría positiva y dispersión elevada con una desv. estándar de alrededor 29,798. La moda de 22,790 indica un prefijo especialmente frecuente. Estos números describen la variabilidad territorial de la base de clientes, como el prefijo es un identificador postal, las métricas reflejan más distribución espacial que un valor continuo.

In [32]:
# Estadística descriptiva del DataFrame products
print("=" * 70)
print("ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: products")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_products = products.select_dtypes(include=[np.number])

# Estadísticas descriptivas básicas
print("\n--- Estadísticas Básicas ---")
print(numeric_columns_products.describe())

# Estadísticas adicionales
print("\n--- Estadísticas Adicionales ---")
print(f"Media:\n{numeric_columns_products.mean()}\n")
print(f"Mediana:\n{numeric_columns_products.median()}\n")
print(f"Moda:\n{numeric_columns_products.mode().iloc[0]}\n")
print(f"Desviación estándar:\n{numeric_columns_products.std()}\n")
print(f"Mínimo:\n{numeric_columns_products.min()}\n")
print(f"Máximo:\n{numeric_columns_products.max()}\n")
print(f"Rango:\n{numeric_columns_products.max() - numeric_columns_products.min()}\n")
print(f"Asimetría (Skewness):\n{numeric_columns_products.skew()}\n")
print(f"Curtosis:\n{numeric_columns_products.kurtosis()}\n")

ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: products

--- Estadísticas Básicas ---
       product_name_lenght  product_description_lenght  product_photos_qty  \
count         32341.000000                32341.000000        32341.000000   
mean             48.476949                  771.495285            2.188986   
std              10.245741                  635.115225            1.736766   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  339.000000            1.000000   
50%              51.000000                  595.000000            1.000000   
75%              57.000000                  972.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      32949.000000       32949.000000       32949.000000   
mean        2276.472488          30.815078          16.937661   
std         4282.038731 

|En products se ve mucha variedad en contenido y dimensiones. Los nombres suelen tener alrededor de 51 caracteres y las descripciones alrededor de 595 (mediana), pero hay fichas muy extensas con un máx de alrededor 3,992, lo que explica la gran dispersión. La cantidad de fotos está concentrada en 1, con algunos productos con muchas imágenes. En lo físico, el peso típico ronda 700 g, mientras que las medidas largo ~25 cm, alto ~13 cm y ancho ~20 cm en mediana muestran colas largas, ya que existen productos muy voluminosos o pesados con un máx de alrededor de 40 kg y dimensiones que superan el metro. En conjunto, las distribuciones de fotos, peso y dimensiones están sesgadas hacia valores altos con pocos casos extremos que elevan medias y rangos.

In [33]:
# Estadística descriptiva del DataFrame sellers
print("=" * 70)
print("ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: sellers")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_sellers = sellers.select_dtypes(include=[np.number])

if numeric_columns_sellers.empty:
    print("\nNo hay columnas numéricas en este DataFrame.")
else:
    # Estadísticas descriptivas básicas
    print("\n--- Estadísticas Básicas ---")
    print(numeric_columns_sellers.describe())
    
    # Estadísticas adicionales
    print("\n--- Estadísticas Adicionales ---")
    print(f"Media:\n{numeric_columns_sellers.mean()}\n")
    print(f"Mediana:\n{numeric_columns_sellers.median()}\n")
    print(f"Moda:\n{numeric_columns_sellers.mode().iloc[0]}\n")
    print(f"Desviación estándar:\n{numeric_columns_sellers.std()}\n")
    print(f"Mínimo:\n{numeric_columns_sellers.min()}\n")
    print(f"Máximo:\n{numeric_columns_sellers.max()}\n")
    print(f"Rango:\n{numeric_columns_sellers.max() - numeric_columns_sellers.min()}\n")
    print(f"Asimetría (Skewness):\n{numeric_columns_sellers.skew()}\n")
    print(f"Curtosis:\n{numeric_columns_sellers.kurtosis()}\n")

ESTADÍSTICA DESCRIPTIVA DEL DATAFRAME: sellers

--- Estadísticas Básicas ---
       seller_zip_code_prefix
count             3095.000000
mean             32291.059451
std              32713.453830
min               1001.000000
25%               7093.500000
50%              14940.000000
75%              64552.500000
max              99730.000000

--- Estadísticas Adicionales ---
Media:
seller_zip_code_prefix    32291.059451
dtype: float64

Mediana:
seller_zip_code_prefix    14940.0
dtype: float64

Moda:
seller_zip_code_prefix    14940
Name: 0, dtype: int64

Desviación estándar:
seller_zip_code_prefix    32713.45383
dtype: float64

Mínimo:
seller_zip_code_prefix    1001
dtype: int64

Máximo:
seller_zip_code_prefix    99730
dtype: int64

Rango:
seller_zip_code_prefix    98729
dtype: int64

Asimetría (Skewness):
seller_zip_code_prefix    0.916092
dtype: float64

Curtosis:
seller_zip_code_prefix   -0.858264
dtype: float64



En sellers, el prefijo postal cubre un rango muy amplio con alrededor de 1,001 a 99,730, lo que apunta a una dispersión geográfica notable. La mediana de alrededor 14,940 queda bastante por debajo de la media de 32,291, con cola hacia prefijos altos con una asimetría positiva y alta dispersión. La moda coincide en 14,940, señal de concentración alrededor de ese código. En síntesis, los vendedores están repartidos por muchas zonas, con algunos grupos más presentes en ciertos prefijos y una cola larga de valores altos que eleva el promedio.

### Verificar valores nulos, registros faltantes y duplicados.

In [34]:
# Verificar valores nulos y duplicados del DataFrame orders
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: orders")
print("=" * 70)

# Valores nulos por columna
null_values_orders = orders.isnull().sum()

# Verificar duplicados
duplicates_orders = orders.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_orders)
print("\nNúmero de registros duplicados:", duplicates_orders)
print("\nTotal de registros:", len(orders))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_orders / len(orders)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: orders

Valores nulos por columna:
 order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

Número de registros duplicados: 0

Total de registros: 99441
Porcentaje de duplicados: 0.00%


En orders no hay duplicados y los nulos se concentran en las fechas intermedias/finales, en order_approved_at tiene 160 faltantes, order_delivered_carrier_date 1,783 y order_delivered_customer_date 2,965. Las columnas order_purchase_timestamp y order_estimated_delivery_date están completas. En conjunto (99,441 registros), la gran mayoría de pedidos cuenta con todos los hitos temporales, y un pequeño porcentaje carece de la entrega real o de pasos intermedios, consistente con órdenes no finalizadas o con estados distintos a “delivered”.

In [35]:
# Verificar valores nulos y duplicados del DataFrame order_items
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: order_items")
print("=" * 70)

# Valores nulos por columna
null_values_order_items = order_items.isnull().sum()

# Verificar duplicados
duplicates_order_items = order_items.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_order_items)
print("\nNúmero de registros duplicados:", duplicates_order_items)
print("\nTotal de registros:", len(order_items))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_order_items / len(order_items)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: order_items

Valores nulos por columna:
 order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

Número de registros duplicados: 0

Total de registros: 112650
Porcentaje de duplicados: 0.00%


In [36]:
# Verificar valores nulos y duplicados del DataFrame order_payments
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: order_payments")
print("=" * 70)

# Valores nulos por columna
null_values_order_payments = order_payments.isnull().sum()

# Verificar duplicados
duplicates_order_payments = order_payments.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_order_payments)
print("\nNúmero de registros duplicados:", duplicates_order_payments)
print("\nTotal de registros:", len(order_payments))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_order_payments / len(order_payments)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: order_payments

Valores nulos por columna:
 order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

Número de registros duplicados: 0

Total de registros: 103886
Porcentaje de duplicados: 0.00%


In [37]:
# Verificar valores nulos y duplicados del DataFrame order_reviews
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: order_reviews")
print("=" * 70)

# Valores nulos por columna
null_values_order_reviews = order_reviews.isnull().sum()

# Verificar duplicados
duplicates_order_reviews = order_reviews.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_order_reviews)
print("\nNúmero de registros duplicados:", duplicates_order_reviews)
print("\nTotal de registros:", len(order_reviews))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_order_reviews / len(order_reviews)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: order_reviews

Valores nulos por columna:
 review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

Número de registros duplicados: 0

Total de registros: 99224
Porcentaje de duplicados: 0.00%


En order_reviews no hay duplicados y los campos clave como review_id, order_id, review_score, fechas están completos. En cambio, los textos están muy incompletos, el título falta en 87,656 reseñas y el mensaje en 58,247, sobre un total de 99,224 registros. Esto deja a review_score como la señal más consistente para resumir satisfacción, mientras que el contenido libre aparece disponible solo en una fracción de casos.

In [38]:
# Verificar valores nulos y duplicados del DataFrame customers
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: customers")
print("=" * 70)

# Valores nulos por columna
null_values_customers = customers.isnull().sum()

# Verificar duplicados
duplicates_customers = customers.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_customers)
print("\nNúmero de registros duplicados:", duplicates_customers)
print("\nTotal de registros:", len(customers))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_customers / len(customers)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: customers

Valores nulos por columna:
 customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

Número de registros duplicados: 0

Total de registros: 99441
Porcentaje de duplicados: 0.00%


In [39]:
# Verificar valores nulos y duplicados del DataFrame products
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: products")
print("=" * 70)

# Valores nulos por columna
null_values_products = products.isnull().sum()

# Verificar duplicados
duplicates_products = products.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_products)
print("\nNúmero de registros duplicados:", duplicates_products)
print("\nTotal de registros:", len(products))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_products / len(products)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: products

Valores nulos por columna:
 product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

Número de registros duplicados: 0

Total de registros: 32951
Porcentaje de duplicados: 0.00%


En products no hay duplicados y la clave product_id está completa. Los nulos se concentran en los campos de contenido product_category_name, product_name_lenght y product_description_lenght, y product_photos_qty con 610 faltantes en cada uno. En cambio, las dimensiones físicas de peso y medidas están casi completas, con solo 2 nulos por campo. Esto sugiere que la ficha técnica básica está bien cubierta, mientras que la información de categoría y detalle de producto puede faltar en una pequeña fracción de casos.

In [40]:
# Verificar valores nulos y duplicados del DataFrame sellers
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: sellers")
print("=" * 70)

# Valores nulos por columna
null_values_sellers = sellers.isnull().sum()

# Verificar duplicados
duplicates_sellers = sellers.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_sellers)
print("\nNúmero de registros duplicados:", duplicates_sellers)
print("\nTotal de registros:", len(sellers))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_sellers / len(sellers)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: sellers

Valores nulos por columna:
 seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

Número de registros duplicados: 0

Total de registros: 3095
Porcentaje de duplicados: 0.00%


In [41]:
# Verificar valores nulos y duplicados del DataFrame cat_tr
print("=" * 70)
print("VALORES NULOS Y DUPLICADOS DEL DATAFRAME: cat_tr")
print("=" * 70)

# Valores nulos por columna
null_values_cat_tr = cat_tr.isnull().sum()

# Verificar duplicados
duplicates_cat_tr = cat_tr.duplicated().sum()

# Mostrar resultados
print("\nValores nulos por columna:\n", null_values_cat_tr)
print("\nNúmero de registros duplicados:", duplicates_cat_tr)
print("\nTotal de registros:", len(cat_tr))
print("Porcentaje de duplicados: {:.2f}%".format((duplicates_cat_tr / len(cat_tr)) * 100))

VALORES NULOS Y DUPLICADOS DEL DATAFRAME: cat_tr

Valores nulos por columna:
 product_category_name            0
product_category_name_english    0
dtype: int64

Número de registros duplicados: 0

Total de registros: 71
Porcentaje de duplicados: 0.00%


### Verificar valores extremos (outliers).

In [42]:
# Verificar outliers del DataFrame order_items
print("=" * 70)
print("OUTLIERS DEL DATAFRAME: order_items")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_order_items = order_items.select_dtypes(include=[np.number])

if not numeric_columns_order_items.empty:
    for column in numeric_columns_order_items.columns:
        # Calcular Z-score
        z_scores = (numeric_columns_order_items[column] - numeric_columns_order_items[column].mean()) / numeric_columns_order_items[column].std()
        
        # Identificar outliers (Z-score > 3 o < -3)
        outliers = abs(z_scores) > 3
        num_outliers = outliers.sum()
        
        print(f"\nColumna: {column}")
        print(f"Número de outliers: {num_outliers}")
        print(f"Porcentaje de outliers: {(num_outliers / len(numeric_columns_order_items)) * 100:.2f}%")
        
        if num_outliers > 0:
            print(f"Valores outliers detectados:\n{numeric_columns_order_items[outliers][column].head(10)}")
else:
    print("\nNo hay columnas numéricas en este DataFrame.")

OUTLIERS DEL DATAFRAME: order_items

Columna: order_item_id
Número de outliers: 1894
Porcentaje de outliers: 1.68%
Valores outliers detectados:
140     4
299     4
329     4
389     4
440     4
557     4
561     4
562     5
976     4
1135    4
Name: order_item_id, dtype: int64

Columna: price
Número de outliers: 1966
Porcentaje de outliers: 1.75%
Valores outliers detectados:
7       810.00
39      809.10
322    1050.61
344     849.00
475     899.00
518    1820.00
700     738.90
865    1240.00
873    1799.00
925     989.00
Name: price, dtype: float64

Columna: freight_value
Número de outliers: 2041
Porcentaje de outliers: 1.81%
Valores outliers detectados:
7      70.75
124    85.97
162    87.28
186    82.86
313    67.69
352    97.56
453    79.94
456    98.13
459    78.63
518    81.62
Name: freight_value, dtype: float64


En order_items aparecen pocos casos extremos en las tres columnas analizadas, con 1.68% en order_item_id, 1.75% en price esto indica importes muy altos respecto al precio típico, y 1.81% en freight_value indicando fletes elevados. Estos resultados confirman una cola larga, ya que la mayoría de filas concentra valores bajos o moderados, pero existe un grupo pequeño con precios y fletes muy altos y pedidos con varios ítems que empujan las medidas basadas en promedios.

In [ ]:
# Verificar outliers del DataFrame order_payments
print("=" * 70)
print("OUTLIERS DEL DATAFRAME: order_payments")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_order_payments = order_payments.select_dtypes(include=[np.number])

if not numeric_columns_order_payments.empty:
    for column in numeric_columns_order_payments.columns:
        # Calcular Z-score
        z_scores = (numeric_columns_order_payments[column] - numeric_columns_order_payments[column].mean()) / numeric_columns_order_payments[column].std()
        
        # Identificar outliers (Z-score > 3 o < -3)
        outliers = abs(z_scores) > 3
        num_outliers = outliers.sum()
        
        print(f"\nColumna: {column}")
        print(f"Número de outliers: {num_outliers}")
        print(f"Porcentaje de outliers: {(num_outliers / len(numeric_columns_order_payments)) * 100:.2f}%")
        
        if num_outliers > 0:
            print(f"Valores outliers detectados:\n{numeric_columns_order_payments[outliers][column].head(10)}")
else:
    print("\nNo hay columnas numéricas en este DataFrame.")

OUTLIERS DEL DATAFRAME: order_payments

Columna: payment_sequential
Número de outliers: 906
Porcentaje de outliers: 0.87%
Valores outliers detectados:
121     4
181     5
285     5
354     8
679     5
815     5
826     4
987     6
1132    7
1200    4
Name: payment_sequential, dtype: int64

Columna: payment_installments
Número de outliers: 341
Porcentaje de outliers: 0.33%
Valores outliers detectados:
195     12
278     13
493     12
1389    12
1446    12
1504    15
2622    12
2970    24
3038    15
3234    11
Name: payment_installments, dtype: int64

Columna: payment_value
Número de outliers: 1803
Porcentaje de outliers: 1.74%
Valores outliers detectados:
160    2027.16
167    1002.73
235    1302.42
247    2288.31
359    1586.47
395     953.71
434    1995.69
695    1783.44
699    1432.92
781    1180.51
Name: payment_value, dtype: float64


En order_payments los casos extremos son pocos, pero visibles en tres frentes, como en el número de pagos por orden ya que aparecen secuencias inusualmente altas con valores 4–8 y más, en cuotas, surgen planes largos (11–24) que se apartan de la concentración en 1–4 cuotas, y en el monto, existe una cola de pagos altos con alrededor de $1,000–$2,300 y superiores, muy por encima del valor típico. En conjunto, el conjunto está dominado por pagos únicos y montos moderados, con una minoría de órdenes que muestran pagos múltiples, financiamientos extensos o importes elevados que explican la asimetría.

In [44]:
# Verificar outliers del DataFrame order_reviews
print("=" * 70)
print("OUTLIERS DEL DATAFRAME: order_reviews")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_order_reviews = order_reviews.select_dtypes(include=[np.number])

if not numeric_columns_order_reviews.empty:
    for column in numeric_columns_order_reviews.columns:
        # Calcular Z-score
        z_scores = (numeric_columns_order_reviews[column] - numeric_columns_order_reviews[column].mean()) / numeric_columns_order_reviews[column].std()
        
        # Identificar outliers (Z-score > 3 o < -3)
        outliers = abs(z_scores) > 3
        num_outliers = outliers.sum()
        
        print(f"\nColumna: {column}")
        print(f"Número de outliers: {num_outliers}")
        print(f"Porcentaje de outliers: {(num_outliers / len(numeric_columns_order_reviews)) * 100:.2f}%")
        
        if num_outliers > 0:
            print(f"Valores outliers detectados:\n{numeric_columns_order_reviews[outliers][column].head(10)}")
else:
    print("\nNo hay columnas numéricas en este DataFrame.")

OUTLIERS DEL DATAFRAME: order_reviews

Columna: review_score
Número de outliers: 0
Porcentaje de outliers: 0.00%


In [45]:
# Verificar outliers del DataFrame customers
print("=" * 70)
print("OUTLIERS DEL DATAFRAME: customers")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_customers = customers.select_dtypes(include=[np.number])

if not numeric_columns_customers.empty:
    for column in numeric_columns_customers.columns:
        # Calcular Z-score
        z_scores = (numeric_columns_customers[column] - numeric_columns_customers[column].mean()) / numeric_columns_customers[column].std()
        
        # Identificar outliers (Z-score > 3 o < -3)
        outliers = abs(z_scores) > 3
        num_outliers = outliers.sum()
        
        print(f"\nColumna: {column}")
        print(f"Número de outliers: {num_outliers}")
        print(f"Porcentaje de outliers: {(num_outliers / len(numeric_columns_customers)) * 100:.2f}%")
        
        if num_outliers > 0:
            print(f"Valores outliers detectados:\n{numeric_columns_customers[outliers][column].head(10)}")
else:
    print("\nNo hay columnas numéricas en este DataFrame.")

OUTLIERS DEL DATAFRAME: customers

Columna: customer_zip_code_prefix
Número de outliers: 0
Porcentaje de outliers: 0.00%


In [46]:
# Verificar outliers del DataFrame products
print("=" * 70)
print("OUTLIERS DEL DATAFRAME: products")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_products = products.select_dtypes(include=[np.number])

if not numeric_columns_products.empty:
    for column in numeric_columns_products.columns:
        # Calcular Z-score
        z_scores = (numeric_columns_products[column] - numeric_columns_products[column].mean()) / numeric_columns_products[column].std()
        
        # Identificar outliers (Z-score > 3 o < -3)
        outliers = abs(z_scores) > 3
        num_outliers = outliers.sum()
        
        print(f"\nColumna: {column}")
        print(f"Número de outliers: {num_outliers}")
        print(f"Porcentaje de outliers: {(num_outliers / len(numeric_columns_products)) * 100:.2f}%")
        
        if num_outliers > 0:
            print(f"Valores outliers detectados:\n{numeric_columns_products[outliers][column].head(10)}")
else:
    print("\nNo hay columnas numéricas en este DataFrame.")

OUTLIERS DEL DATAFRAME: products

Columna: product_name_lenght
Número de outliers: 167
Porcentaje de outliers: 0.51%
Valores outliers detectados:
447     17.0
811     15.0
856     16.0
864     10.0
946     17.0
1143    15.0
1468    16.0
1632    16.0
1881    14.0
2305    16.0
Name: product_name_lenght, dtype: float64

Columna: product_description_lenght
Número de outliers: 696
Porcentaje de outliers: 2.11%
Valores outliers detectados:
15     3021.0
50     3218.0
176    2913.0
231    3830.0
272    2819.0
322    3354.0
357    2959.0
382    3492.0
508    3939.0
509    2952.0
Name: product_description_lenght, dtype: float64

Columna: product_photos_qty
Número de outliers: 506
Porcentaje de outliers: 1.54%
Valores outliers detectados:
31      9.0
35      9.0
118     9.0
315    12.0
316    12.0
480    10.0
548    11.0
586    11.0
874    10.0
967    17.0
Name: product_photos_qty, dtype: float64

Columna: product_weight_g
Número de outliers: 961
Porcentaje de outliers: 2.92%
Valores outliers de

En products los outliers aparecen en varias dimensiones del catálogo. En contenido, hay nombres inusualmente cortos con 0.5% y descripciones muy extensas con 2.1% que llegan a más de 3,900 caracteres. En fotos, la mayoría de productos tiene 1 imagen, pero el grupo atípico acumula 9–17 fotos. En lo físico, se observan pesos altos con 2.9% de casos extremos, varios entre 15–27 kg y dimensiones grandes, largos y alturas cercanos a 100–105 cm, y anchos de 60–105 cm. En conjunto, el catálogo muestra una base concentrada en fichas y productos típicos, con una cola pequeña de artículos muy voluminosos o pesados, o con descripciones y fotos inusualmente extensas que explican la dispersión.

In [47]:
# Verificar outliers del DataFrame sellers
print("=" * 70)
print("OUTLIERS DEL DATAFRAME: sellers")
print("=" * 70)

# Seleccionar solo columnas numéricas
numeric_columns_sellers = sellers.select_dtypes(include=[np.number])

if not numeric_columns_sellers.empty:
    for column in numeric_columns_sellers.columns:
        # Calcular Z-score
        z_scores = (numeric_columns_sellers[column] - numeric_columns_sellers[column].mean()) / numeric_columns_sellers[column].std()
        
        # Identificar outliers (Z-score > 3 o < -3)
        outliers = abs(z_scores) > 3
        num_outliers = outliers.sum()
        
        print(f"\nColumna: {column}")
        print(f"Número de outliers: {num_outliers}")
        print(f"Porcentaje de outliers: {(num_outliers / len(numeric_columns_sellers)) * 100:.2f}%")
        
        if num_outliers > 0:
            print(f"Valores outliers detectados:\n{numeric_columns_sellers[outliers][column].head(10)}")
else:
    print("\nNo hay columnas numéricas en este DataFrame.")

OUTLIERS DEL DATAFRAME: sellers

Columna: seller_zip_code_prefix
Número de outliers: 0
Porcentaje de outliers: 0.00%
